pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -560,6 +562,46 @@ func getNodeSecurityGroupIDForLB(compute *gophercloud.ServiceClient, nodes []*v1
 	return nodeSecurityGroupIDs.List(), nil
 }
 
+// getFloatingNetworkIdForLB returns a floating-network-id for cluster.
+func getFloatingNetworkIdForLB(client *gophercloud.ServiceClient) (string, error) {
+	var FloatingNetworkIds []string
+
+	type NetworkWithExternalExt struct {
+		networks.Network
+		external.NetworkExternalExt
+	}
+	var externalNetwork []NetworkWithExternalExt
+	err := networks.List(client, networks.ListOpts{}).EachPage(func(page pagination.Page) (bool, error) {
+		err := networks.ExtractNetworksInto(page, &externalNetwork)
+		if err != nil {
+			return false, err
+		}
+
+		for _, externalNet := range externalNetwork {
+			if externalNet.External == true {
+				FloatingNetworkIds = append(FloatingNetworkIds, externalNet.ID)
+			}
+		}
+
+		if len(FloatingNetworkIds) > 1 {
+			return false, ErrMultipleResults
```

Comment:
Hrm.  If we have multiple external networks, is it reasonable to just pick any one of them (eg the "first" one) - assuming we don't have more explicit configuration from the user?

Or if there is some sort of recognisable error when we get it wrong, perhaps we should try to use each of them in turn when setting up the LB?

  (trying to make this logic "just work" in as many cases as possible)

Reply:
Yeah, makes sense, if there are multiple external networks, I just return err, it is bad.
I update it soon.